src/cascadia/ElevateShim/elevate-shim.cpp

```diff
@@ -42,38 +42,26 @@ int __stdcall wWinMain(HINSTANCE, HINSTANCE, LPWSTR, int)
     //    cmd:    {same path as this binary}\WindowsTerminal.exe
     //    params: new-tab -p {guid}
 
-    // see if we're a store app we can invoke with shell:AppsFolder
-    std::wstring appUserModelId;
-    const auto result = wil::AdaptFixedSizeToAllocatedResult<std::wstring, APPLICATION_USER_MODEL_ID_MAX_LENGTH>(
-        appUserModelId, [&](PWSTR value, size_t valueLength, gsl::not_null<size_t*> valueLengthNeededWithNull) noexcept -> HRESULT {
-            UINT32 length = gsl::narrow_cast<UINT32>(valueLength);
-            const LONG rc = GetCurrentApplicationUserModelId(&length, value);
-            switch (rc)
-            {
-            case ERROR_SUCCESS:
-                *valueLengthNeededWithNull = length;
-                return S_OK;
-
-            case ERROR_INSUFFICIENT_BUFFER:
-                *valueLengthNeededWithNull = length;
-                return S_FALSE; // trigger allocation loop
-
-            case APPMODEL_ERROR_NO_APPLICATION:
-                return E_FAIL; // we are not running as a store app
+    std::wstring cmd;
 
-            default:
-                return E_UNEXPECTED;
-            }
-        });
-    LOG_IF_FAILED(result);
-
-    std::wstring cmd = {};
-    if (result == S_OK && appUserModelId.length() > 0)
+    // scenario #1
     {
-        // scenario #1
-        cmd = L"shell:AppsFolder\\" + appUserModelId;
+#pragma warning(suppress : 26494) // Variable 'buffer' is uninitialized. Always initialize an object (type.5).
+        wchar_t buffer[APPLICATION_USER_MODEL_ID_MAX_LENGTH];
+        // "On input, the size of the applicationUserModelId buffer, in wide characters."
+        UINT32 length = APPLICATION_USER_MODEL_ID_MAX_LENGTH;
+
+        const auto result = GetCurrentApplicationUserModelId(&length, &buffer[0]);
+        if (result == ERROR_SUCCESS)
+        {
+            cmd.append(L"shell:AppsFolder\\");
+            // "On success, the size of the buffer used, including the null terminator."
+            // --> Remove the null terminator
+            cmd.append(&buffer[0], length - 1);
+        }

```

Comment:
`GetCurrentApplicationUserModelId` has a publicly documented `APPLICATION_USER_MODEL_ID_MAX_LENGTH` so there's no need for dynamic allocations.

While a situation like `MAX_PATH` is theoretically possible, I don't think it's likely, given that file paths are user-configurable and app IDs aren't. In any case, we're checking the `result` properly, so the worst that happens is that it'll just not use the `shell:AppsFolder` trick (which it honestly shouldn't have to do in the first place, given that this more like a really annoying bug in `ShellExecuteExW` if anything).